docs/guide/ubuntu-setup.md

```diff
@@ -0,0 +1,216 @@
+
+## Deploying a site for the first time (each time you add a new site)
+
+> You can run more than one site on a server. For security and performance, you might prefer to run them on separate servers in production.
+
+* If you didn't already, `ssh` to the `ubuntu` user on your server. Then run `sudo su - nodeapps` to switch users.
+
+* We'll `git clone` our project here as a simple way to deploy it. We'll use the `a3-boilerplate` project as an example but you should use your own:
+
+```
+git clone https://github.com/apostrophecms/a3-boilerplate
+cd a3-boilerplate
+npm install
+```
+
+* Now build production assets:
+
+```
+npm run build
+```
+
+Next we instruct `pm2` to launch the site and to keep it running. Substitute the shortname of your own project for `a3-boilerplate` below.
+
+```
+pm2 --name=a3-boilerplate start npm -- run serve
+pm2 save
+```
+
+The second command saves our `pm2` configuration for future reboots.
+
+At this point Apostrophe is running on port `3000`, but we need to configure `nginx`
+as a proxy server to handle HTTP and HTTPS connections on port `80` and `443` and
+forward them.
+
+## Adding your site to nginx (each time you add a new site)
+
+* ssh to the `ubuntu` user shell, where you have sudo access, and create 
+`/etc/nginx/conf.d/your-project-shortname-here.conf`, replacing `your.host.name` with the DNS hostname you have pointed to your server's IP address for your site. Use this command:
+
+```
+sudo nano /etc/nginx/conf.d/your-project-shortname-here.conf
+```
+
+In the editor, paste the following and make the appropriate replacement for `server_name`:
+
+```
+server {
+  listen *:80;
+  server_name your.host.name;
+  location @proxy {
+    proxy_pass http://localhost:3000;
+    proxy_set_header Host $host;
+    proxy_set_header X-Real-IP $remote_addr;
+    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+    proxy_set_header X-Forwarded-Proto $scheme;
+  }
+  location / {
+    root /home/nodeapps/a3-boilerplate/public;
+    try_files $uri @proxy;
+    expires 7d;
+  }
+}
+```
+
+To save, press Control-X, then follow the steps to confirm.
+
+The `proxy_set_header` statements pass information to Apostrophe so it can still see
+the visitor's IP address.
+
+The `root` and `try_files` statements let nginx serve static files directly, for
+the best speed; if the URL isn't a static file, it is passed to Apostrophe. `expires 7d` allows the browser to cache the static files, for performance.
+
+You'll want to add SSL for HTTPS connections too. For that, follow the [Certbot documentation](https://certbot.eff.org/lets-encrypt/ubuntufocal-nginx.html).
+Certbot will make the necessary nginx configuration changes for you.
+
+* Now instruct `nginx` to restart:
+
+```
+sudo systemctl reload nginx
+```
+
+Now your site is up! Visit `http://your.host.name` to see it. If you didn't add it to
+your DNS yet, or it hasn't propagated, you won't be able to reach it yet. You can try `http://your.server.ip.address`
```

Comment:
@boutell Are you sure about the IP address with the config here? I'm going through it and not accessing via the IP.

Reply:
The next step is also suggesting there's no db yet, which means we shouldn't have anything to see.

Reply:
A hostname does work, of course

Reply:
What do you get with the IP? Maybe there's a default site in the nginx conf and it would be simplest to streamline by removing the IP option from the article...

Reply:
i removed it in my edit.